### PR TITLE
Create XOR_linked_list.c

### PR DESCRIPTION
**Pull request by SahilK-027**
#### Description of Change
Create  XOR_linked_list.c

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: data_structures/linked_list/XOR_linked_list.c [hunk 1]
```diff
@@ -0,0 +1,221 @@
+#include <assert.h> /// to verify assumptions made by the program and print a diagnostic message if this assumption is false.
+#include <inttypes.h> /// to provide a set of integer types with universally consistent definitions that are operating system-independent
+#include <stdio.h> /// for IO operations
+#include <stdlib.h> /// for including functions involving memory allocation such as `malloc`
+#include <stdint.h> /// to ensure that the integer types such as uintptr_t are properly defined.
+
+/**
```
Comment (realstealthninja) [inline data_structures/linked_list/XOR_linked_list.c:1]: this should go before the inclusion of files

File: data_structures/linked_list/XOR_linked_list.c [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+ * @brief
+ * [XOR List](https://en.wikipedia.org/wiki/XOR_linked_list) combines the properties of a 
+ * circular doubly linked list and XOR pointers to save memory and enhance performance. 
+ * Each node contains an XOR of the addresses of the previous and next nodes.
+ *
+ * In this implementation, functions to insert at the head, delete the first node, display list, get
+ * list size, and get element at index functions are coded.
+ *
+ * @author [Sahil Kandhare](https://github.com/SahilK-027)
+ */
```
Comment (realstealthninja) [inline data_structures/linked_list/XOR_linked_list.c:13]:
```suggestion
```

File: data_structures/linked_list/XOR_linked_list.c [hunk 3]
```diff
@@ -0,0 +1,221 @@
+#include <assert.h> /// to verify assumptions made by the program and print a diagnostic message if this assumption is false.
```
Comment (realstealthninja) [inline data_structures/linked_list/XOR_linked_list.c]:
```suggestion
/**
 * @file
 *
 * @brief
 * [XOR List](https://en.wikipedia.org/wiki/XOR_linked_list) combines the properties of a 
 * circular doubly linked list and XOR pointers to save memory and enhance performance. 
 * Each node contains an XOR of the addresses of the previous and next nodes.
 *
 * In this implementation, functions to insert at the head, delete the first node, display list, get
 * list size, and get element at index functions are coded.
 *
 * @author [Sahil Kandhare](https://github.com/SahilK-027)
 */
#include <assert.h> /// to verify assumptions made by the program and print a diagnostic message if this assumption is false.
```

**Review by realstealthninja — CHANGES_REQUESTED** [review]
Very nicely done

**Comment by SahilK-027** [comment]
Are the changes fine?

**Review by realstealthninja — APPROVED** [review]
LGTM

**Pull request closed without merging**